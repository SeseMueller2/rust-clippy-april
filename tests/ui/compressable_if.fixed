#![warn(clippy::compressable_if)]
#![allow(clippy::needless_if)]

fn main() -> Result<(), ()> {
    let test = 1;
    if test == 1 || test == 3 {
        println!("odd");
    } 
    if test == 2 {
        println!("even");
    }
    

    if test == 1 || test == 3 {} 
    

    // Negative Tests

    if test == 1 {
        println!("test is 1, so odd");
    }
    if test == 2 {
        println!("test is 2, so even");
    }
    if test == 3 {
        println!("test is 3, so odd");
    }

    Ok(())
}
