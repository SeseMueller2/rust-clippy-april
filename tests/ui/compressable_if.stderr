error: these if blocks have the same content and no side effects
  --> tests/ui/compressable_if.rs:16:5
   |
LL |     if test == 1 {}
   |     ^^^^^^^^^^^^^^^
LL |     if test == 3 {}
   |     ^^^^^^^^^^^^^^^
   |
   = help: consider merging them into a single if block with a combined condition
   = note: `-D clippy::compressable-if` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::compressable_if)]`
help: merge these if blocks
   |
LL ~     if test == 1 || test == 3 {} 
LL ~     
   |

error: these if blocks have the same content and no side effects
  --> tests/ui/compressable_if.rs:6:5
   |
LL | /     if test == 1 {
LL | |         println!("odd");
LL | |     }
   | |_____^
...
LL | /     if test == 3 {
LL | |         println!("odd");
LL | |     }
   | |_____^
   |
   = help: consider merging them into a single if block with a combined condition
help: merge these if blocks
   |
LL ~     if test == 1 || test == 3 {
LL +         println!("odd");
LL +     } 
LL |     if test == 2 {
LL |         println!("even");
LL |     }
LL ~     
   |

error: aborting due to 2 previous errors

